src/analyses/commonPriv.ml

```diff
@@ -197,6 +197,22 @@ struct
     | _ -> false
 end
 
+module ThreadNotStartedDigest:Digest =
+struct
+  include ThreadIdDomain.ThreadLifted
+
+  module TID = ThreadIdDomain.Thread
+
+  let current (ask: Q.ask) =
+    ThreadId.get_current ask
+
+  let accounted_for (ask: Q.ask) ~(current: t) ~(other: t) =
+    match current, other with
+    | `Lifted current, `Lifted other ->
+      MHP.definitely_not_started (current, ask.f Q.CreatedThreads) other
+    | _ -> false
+end
```

Comment:
Yes, but things such as not reading from yourself and not reading from joined threads goes beyond what the digests can do, and relies on the analysis taking care of it in some way.

Consider e.g. reading from a thread that has been joined: It is possible that this thread held the mutex last before me, and thus the digests are `compatible`.
It is only because the analyses also read from some L that this contribution is `accounted_for`.

Reply:
In more-traces this is the difference between
$$[[lock(a)]]^\sharp_\mathcal{A}(A_0,A_1) = \emptyset$$

and

$$\textsf{acc} A_0 A_1$$

Reply:
Thanks for the explanation. I guess the coupling is kind of implicit here: this `accounted_for` assumes that the privatization actually accounts for things in its `thread_join`.